homeassistant/components/automation/zone.py

```diff
@@ -20,7 +21,7 @@
 TRIGGER_SCHEMA = vol.Schema({
     vol.Required(CONF_PLATFORM): 'zone',
     vol.Required(CONF_ENTITY_ID): cv.entity_ids,
-    vol.Required(CONF_ZONE): cv.entity_id,
+    vol.Required(CONF_ZONE): cv.zone_id,
```

Comment:
There is no need to add this to `config_validation.py` as this is not something that will most likely to be reused. Just add this to `automation/zone.py` instead:

```python
vol.Required(CONF_ZONE): vol.Any('*', cv.entity_id)
```